### DISCUSSION TITLE
Save Draft Meetings.

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#954] [Q&A]
How can I ensure that draft meetings do not get over written if I do an import of a CSV file?

Background: We have a few seasonal meetings that are put in draft mode that I’d like to preserve. For example we have outdoor meetings during the summer months. These meetings are not included in a CSV export. But I find when I do the CSV import that everything gets over written and no longer have these draft meetings. 

Initially I thought if I put them in the trash and did not empty the trash, that these meetings would stay even with doing a CSV import. But that turned out not to be the case either.

**Comment by joshreisner** [comment]
hi Viv, could you tell us more about this use-case? what percentage of the time do you update on the site vs in a spreadsheet?

i think we kind of expected people to manage only on the site or in a spreadsheet. 

one approach you could take would be to move the seasonal meetings to a different tab in your worksheet for the winter and then copy them back in the spring.

if we were to solve this in TSML what would be the solution? either 

1. make it so "delete all meetings, locations, regions, etc" doesn't include drafts
2. add a new option: delete all meetings, locations, regions, etc BUT NOT DRAFTS!"
3. export the publishing status as a column in the CSV aka "draft" or "published" so when reimporting they were given the right status

**Reply by Grateful-Viv** [reply]
The percentage is mixed. Most of the time editing is done right on the site. However a few times this year, there were some wholesale changes done that using the import CSV made more sense. The last was when all online meetings with "tc" status were stripped of that type. Because the location field is tied to the address field, if I edited one meeting on the site to have the location be "online meeting" and the address to "city, state", then all meetings with either the same address (city, state) would change the location to "online meeting" or vice versa. I found that I needed to create a unique identifier for the combination of location AND address, if I wanted them to be noted separately. 

Doing these changes on the site proved cumbersome, so I did it in the spreadsheet and then uploaded it using the "delete all meetings, locations, groups, districts, and regions" option under import.

I have actually saved a worksheet since then that does contain the seasonal meeting data. Then will use that info to enter them again in case the draft meetings get deleted, but if there's a possibility of doing any one of the three options you present, then that would be great. I am partial to adding new import option "delete all meetings, locations, groups, districts, and regions, except draft meetings" would suffice. Is this possible? Is it too much trouble? 

Thanks for considering all this.

**Reply by joshreisner** [reply]
Ok thanks. I'll let the TSML maintainers weigh in on this one…
